core/src/main/java/org/kiwix/kiwixmobile/core/search/SearchActivity.java

```diff
@@ -285,24 +266,17 @@ private void resetAdapter() {
     searchPresenter.getRecentSearches();
   }
 
-  private ArrayAdapter<String> getDefaultAdapter() {
-    return new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1) {
-      @NonNull
-      @Override
-      public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
-        View row;
-
-        if (convertView == null) {
-          row = LayoutInflater.from(parent.getContext())
-            .inflate(android.R.layout.simple_list_item_1, null);
-        } else {
-          row = convertView;
-        }
+  private DefaultAdapter getDefaultAdapter() {
+    return new DefaultAdapter(this, new DefaultAdapter.CLickListener() {
```

Comment:
Another option here is to make SearchActivity implement DefaultAdapter.CLickListener and construct DefaultAdapter like `new DefaultAdapter(this, this)`

This change isn't particularly needed but it keeps this class looking more like it used to. I'll leave it to your discrection